tsl/test/src/remote/connection.c

```diff
@@ -244,19 +240,19 @@ ts_test_remote_connection(PG_FUNCTION_ARGS)
 }
 
 pid_t
-remote_connecton_get_remote_pid(TSConnection *conn)
+remote_connecton_get_remote_pid(const TSConnection *conn)
 {
 	PGresult *res;
 	char *pid_string;
 	unsigned long pid_long;
 
-	res = PQexec(remote_connection_get_pg_conn(conn), sql_get_backend_pid);
+	res = PQexec(remote_connection_get_pg_conn(conn), "SELECT pg_backend_pid()");
```

Comment:
Depending on how this function is used, there is a possibility that the connection is not to the local host and will return a pid on a different machine. Trying to kill the pid in that case will at best lead to strange errors, at worst kill a process that shouldn't be killed.

One approach to preventing this is to check that the connection is to the local machine and error out if it is not.

Reply:
This is only used in tests and we control how it is used. Not sure it is worth doing these checks, and this function is not available in the general code base.